File: project2/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private TextView tvTime;
    private Button btnStart, btnPause, btnReset;

    private Handler handler = new Handler();
    private long startTime, timeInMillis;
    private boolean isRunning;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvTime = findViewById(R.id.tvTime);
        btnStart = findViewById(R.id.btnStart);
        btnPause = findViewById(R.id.btnPause);
        btnReset = findViewById(R.id.btnReset);

        btnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startTime = System.currentTimeMillis();
                handler.postDelayed(updateTimerRunnable, 0);
                isRunning = true;
                btnStart.setEnabled(false);
                btnPause.setEnabled(true);
                btnReset.setEnabled(false);
            }
        });

        btnPause.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                timeInMillis += System.currentTimeMillis() - startTime;
                handler.removeCallbacks(updateTimerRunnable);
                isRunning = false;
                btnStart.setEnabled(true);
                btnPause.setEnabled(false);
                btnReset.setEnabled(true);
            }
        });

        btnReset.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                timeInMillis = 0;
                updateTimerText(0);
                btnReset.setEnabled(false);
            }
        });
    }

    private Runnable updateTimerRunnable = new Runnable() {
        @Override
        public void run() {
            if (isRunning) {
                long elapsedMillis = System.currentTimeMillis() - startTime + timeInMillis;
                updateTimerText(elapsedMillis);
                handler.postDelayed(this, 10);
            }
        }
    };

    private void updateTimerText(long elapsedMillis) {
        int minutes = (int) (elapsedMillis / 60000);
        int seconds = (int) (elapsedMillis / 1000) % 60;
        int millis = (int) (elapsedMillis % 1000);

        String time = String.format("%02d:%02d:%03d", minutes, seconds, millis);
        tvTime.setText(time);
    }
}

File: app/src/main/java/com/example/myapplication/MainActivity.kt
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.example.myapplication.R

class MainActivity : AppCompatActivity() {

    private lateinit var btnAdd: Button
    private lateinit var btnSub: Button
    private lateinit var btnMulti: Button
    private lateinit var btnDiv: Button
    private lateinit var edit1: EditText
    private lateinit var edit2: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Initialize the views
        btnAdd = findViewById(R.id.btnAdd)
        btnSub = findViewById(R.id.btnSub)
        btnMulti = findViewById(R.id.btnMulti)
        btnDiv = findViewById(R.id.btnDiv)
        edit1 = findViewById(R.id.edit1)
        edit2 = findViewById(R.id.edit2)

        btnAdd.setOnClickListener {
            val no1 = edit1.text.toString().toIntOrNull()
            val no2 = edit2.text.toString().toIntOrNull()

            if (no1 != null && no2 != null) {
                val sum = no1 + no2
                Toast.makeText(this, "The Addition is $sum", Toast.LENGTH_LONG).show()
            } else {
                Toast.makeText(this, "Please fill the required blanks.", Toast.LENGTH_LONG).show()
            }
        }

        btnSub.setOnClickListener {
            val no1 = edit1.text.toString().toIntOrNull()
            val no2 = edit2.text.toString().toIntOrNull()

            if (no1 != null && no2 != null) {
                val sub = if (no1 > no2) no1 - no2 else no2 - no1
                Toast.makeText(this, "The Subtraction is $sub", Toast.LENGTH_LONG).show()
            } else {
                Toast.makeText(this, "Please fill the required blanks.", Toast.LENGTH_LONG).show()
            }
        }

        btnMulti.setOnClickListener {
            val no1 = edit1.text.toString().toIntOrNull()
            val no2 = edit2.text.toString().toIntOrNull()

            if (no1 != null && no2 != null) {
                val multi = no1 * no2
                Toast.makeText(this, "The Multiplication is $multi", Toast.LENGTH_LONG).show()
            } else {
                Toast.makeText(this, "Please fill the required blanks.", Toast.LENGTH_LONG).show()
            }
        }

        btnDiv.setOnClickListener {
            val no1 = edit1.text.toString().toIntOrNull()
            val no2 = edit2.text.toString().toIntOrNull()

            if (no1 != null && no2 != null) {
                if (no2 == 0) {
                    Toast.makeText(this, "A number cannot be divided by zero.", Toast.LENGTH_LONG).show()
                } else {
                    val div = no1 / no2
                    Toast.makeText(this, "The Division is $div", Toast.LENGTH_LONG).show()
                }
            } else {
                Toast.makeText(this, "Please fill the required blanks.", Toast.LENGTH_LONG).show()
            }
        }
    }
}
